### DISCUSSION TITLE
info related to who send event CEventNetworkEntityDamage in get_network_event_data

### DISCUSSION THREAD

**Discussion by lonelybud** [YimMenu/YimMenu#2693] [General]
this can help get some data to find out who blamed you for killing other player.

For example when the kill happens we can know who sent event and there distance from dead player. Even though their distance can be close to dead guy still we can get idea who is sending those events.

**Comment by Yimura** [comment]
Have you also checked if this works properly for owned explosions? As these can be sent by anyone in the session and still be caused by another player.

**Reply by lonelybud** [reply]
i am sorry but i dont understand.

**Reply by lonelybud** [reply]
i think this is a major concern. one should find out a way to know blame kills.

**Reply by Yimura** [reply]
You know about those bombs that only trigger when you come near them, right? Well if someone puts them there but then leaves the area (say he goes to the other side of the map). If you go over it, what will happen is that your client (game) explodes it for them and awards them them the kill. This means that your client created the explosion and also sent out the network event of the explosion even though the bomb was put down by someone else.

This is what makes it hard to detect if someone abuses the native to trigger explosions trying to blame other players for the explosion. As you cannot always trust the person sending out the event is that one that forced the explosion (aka the blamed kill). This is why you need to figure out from the network data within it who actually sent the event.

**Reply by Yimura** [reply]
YimMenu has this detection built-in however it false positives like crazy and should NEVER be relied on.

**Reply by lonelybud** [reply]
i thought that people who died will send those events and those who were not near the explosion area will not. That way we can figure out who did not sent the event and narrow down the players who blamed us.
